src/main/webapp/definitions/alfresco-core.yaml

```diff
@@ -2484,7 +2491,96 @@ paths:
           description: Unexpected error
           schema:
             $ref: '#/definitions/Error'
+  '/nodes/{nodeId}/calculateSize':
+    post:
+      tags:
+        - nodes
+      summary: Calculate a folder size
+      description: |
+        
+        This may be executed against a node specified by **nodeId** and this nodeId must be a valid node ID.
+        Furthermore, this POST endpoint will be executed **asynchronously** with a `202` HTTP response signifying that
+        the request has been accepted successfully.
+        
+        The response body will include the ```nodeId``` of the pending action, which can be used in a
+        **GET/calculateSize** endpoint to check if the action's status has been completed, at which point the result will be returned, comprising ```the size of the node in bytes```.
+        
+        Essentially, ```maxItems``` ***parameter*** is utilized for pagination purposes.
+        
+        By default, **results are limited to the first 100.**
+        Results can be restricted using "paging". For example:
+        
+        ```JSON
+        "paging": {
+          "maxItems": "50",
+          "skipCount": "28"
+        }
+        ```
+        This assists in handling scenarios where large files or subfolders exist within a folder, allowing the folder size determination mechanism to function in chunks.
 
+      operationId: calculateSize
+      produces:
+        - application/json
+      parameters:
+        - $ref: '#/parameters/nodeTargetIdParam'
+        - $ref: '#/parameters/maxItemsForSizeParam'
+      responses:
+        '202':
+          description: Request Accepted
+          schema:
+            $ref: '#/definitions/sizeAcknowledgedResponse'
+        '400':
+          description: |
+            Invalid parameter: value of **nodeId** is invalid
+        '401':
+          description: Authentication failed
+        '403':
+          description: Current user does not have permission for **nodeId**
```

Comment:
What happens if the current user has access to `nodeId` but not to a child folder?

Reply:
As we are executing an Alfresco FTS query through the Search Subsystem, we are only receiving all linked **nodeIds** at once. As a result, we are just restricting permission validation to the provided node.

Reply:
Ok - so you should either update the documentation to say:
* The calculation is run as system and so will include the size of documents that the user does not have access to.
OR
* The calculation is restricted to documents that the user has access to and so the true folder size may be larger than returned.